src/um/msxml.rs

```diff
@@ -0,0 +1,852 @@
+    ) -> HRESULT,
+    fn substringData(
+        offset: c_long,
+        count: c_long,
+        data: *mut BSTR,
+    ) -> HRESULT,
+    fn appendData(
+        data: BSTR,
+    ) -> HRESULT,
+    fn insertData(
+        offset: c_long,
+        data: BSTR,
+    ) -> HRESULT,
+    fn deleteData(
+        offset: c_long,
+        count: c_long,
+    ) -> HRESULT,
+    fn replaceData(
+        offset: c_long,
+        count: c_long,
+        data: BSTR,
+    ) -> HRESULT,
+});
+RIDL!(#[uuid(0x2933bf85, 0x7b36, 0x11d2, 0xb2, 0x0e, 0x00, 0xc0, 0x4f, 0x98, 0x3e, 0x60)]
+interface IXMLDOMAttribute(IXMLDOMAttributeVtbl): IXMLDOMNode(IXMLDOMNodeVtbl) {
+    fn get_name(
+        attributeName: *mut BSTR,
+    ) -> HRESULT,
+    fn get_value(
+        attributeValue: *mut VARIANT,
+    ) -> HRESULT,
+    fn put_value(
+        attributeValue: VARIANT,
+    ) -> HRESULT,
+});
+RIDL!(#[uuid(0x2933bf86, 0x7b36, 0x11d2, 0xb2, 0x0e, 0x00, 0xc0, 0x4f, 0x98, 0x3e, 0x60)]
+interface IXMLDOMElement(IXMLDOMElementVtbl): IXMLDOMNode(IXMLDOMNodeVtbl) {
+    fn get_tagName(
+        tagName: *mut BSTR,
+    ) -> HRESULT,
+    fn getAttribute(
+        name: BSTR,
+        value: *mut VARIANT,
+    ) -> HRESULT,
+    fn setAttribute(
+        name: BSTR,
+        value: VARIANT,
+    ) -> HRESULT,
+    fn removeAttribute(
+        name: BSTR,
+    ) -> HRESULT,
+    fn getAttributeNode(
+        name: BSTR,
+        attributeNode: *mut *mut IXMLDOMAttribute,
+    ) -> HRESULT,
+    fn setAttributeNode(
+        DOMAttribute: *mut IXMLDOMAttribute,
+        attributeNode: *mut *mut IXMLDOMAttribute,
+    ) -> HRESULT,
+    fn removeAttributeNode(
+        DOMAttribute: *mut IXMLDOMAttribute,
+        attributeNode: *mut *mut IXMLDOMAttribute,
+    ) -> HRESULT,
+    fn getElementsByTagName(
+        tagName: BSTR,
+        resultList: *mut *mut IXMLDOMNodeList,
+    ) -> HRESULT,
+    fn normalize() -> HRESULT,
+});
+RIDL!(#[uuid(0x2933bf87, 0x7b36, 0x11d2, 0xb2, 0x0e, 0x00, 0xc0, 0x4f, 0x98, 0x3e, 0x60)]
+interface IXMLDOMText(IXMLDOMTextVtbl): IXMLDOMCharacterData(IXMLDOMCharacterDataVtbl) {
+    fn splitText(
+        offset: c_long,
+        rightHandTextNode: *mut *mut IXMLDOMText,
+    ) -> HRESULT,
+});
+RIDL!(#[uuid(0x2933bf88, 0x7b36, 0x11d2, 0xb2, 0x0e, 0x00, 0xc0, 0x4f, 0x98, 0x3e, 0x60)]
+interface IXMLDOMComment(IXMLDOMCommentVtbl): IXMLDOMCharacterData(IXMLDOMCharacterDataVtbl) {
+});
+RIDL!(#[uuid(0x2933bf89, 0x7b36, 0x11d2, 0xb2, 0x0e, 0x00, 0xc0, 0x4f, 0x98, 0x3e, 0x60)]
+interface IXMLDOMProcessingInstruction(IXMLDOMProcessingInstructionVtbl):
+    IXMLDOMNode(IXMLDOMNodeVtbl) {
+    fn get_target(
+        name: *mut BSTR,
+    ) -> HRESULT,
+    fn get_data(
+        value: *mut BSTR,
+    ) -> HRESULT,
+    fn put_data(
+        value: BSTR,
+    ) -> HRESULT,
+});
+RIDL!(#[uuid(0x2933bf8a, 0x7b36, 0x11d2, 0xb2, 0x0e, 0x00, 0xc0, 0x4f, 0x98, 0x3e, 0x60)]
+interface IXMLDOMCDATASection(IXMLDOMCDATASectionVtbl): IXMLDOMText(IXMLDOMTextVtbl) {
+});
```

Comment:
If the body is empty, put the closing brace on the same line as the opening brace.